modules/ndu/src/backend/ndu-engine.ts

```diff
@@ -12,6 +12,7 @@ import { getTriggerId, stringToVec, WfIdToTopic } from './utils'
 const debug = DEBUG('ndu').sub('processing')
 
 export const DEFAULT_MIN_CONFIDENCE = 0.1
+const fakeConditions = ['on_active_workflow', 'on_active_topic']
```

Comment:
why make fake conditions and not use real conditions for this ? From what I see, we can easily write this in a condition

Reply:
These are special conditions. If you have activeWorkflow, and the user is not on the workflow, then you want to skip processing the whole trigger. If you use a normal condition, then it would still process all the conditions associated with the trigger, and would add the trigger to the ndu's predictions.

The alternative would be to loop through all the trigger's conditions to see if there is this special condition, then filter it out. We chose activeWorkflow a long time ago